src/transformStateWithClones.js

```diff
@@ -1,13 +1,40 @@
 'use strict';
-
-/**
+/*
  * @param {Object} state
  * @param {Object[]} actions
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  const updateState = { ...state };
+  const updateActions = [];
+
+  for (const key of actions) {
+    switch (key.type) {
+      case 'addProperties':
+        Object.assign(updateState, key.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const item of key.keysToRemove) {
+          delete updateState[item];
+        }
+        break;
+
+      case 'clear':
+        for (const a in updateState) {
+          delete updateState[a];
```

Comment:
Instead of using a for-in loop to clear the properties of `updateState`, you can simply set it to an empty object `{}` to achieve the same effect more efficiently.